From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paolo Sabatino <paolo.sabatino@gmail.com>
Date: Wed, 20 Aug 2025 21:40:48 +0200
Subject: [ARCHEOLOGY] workaround for rockchip gigabit ethernet not able to
 init dma

Temporary patch to workaround a DMA reset issue with rockchip
devices experienced at least on rk3288 and rk3328 with a message
like this:

rk_gmac-dwmac ff290000.ethernet end0: Failed to reset the dma
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index 111111111111..222222222222 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -3057,8 +3057,8 @@ static int stmmac_init_dma_engine(struct stmmac_priv *priv)
 
 	ret = stmmac_reset(priv, priv->ioaddr);
 	if (ret) {
-		netdev_err(priv->dev, "Failed to reset the dma\n");
-		return ret;
+		netdev_warn(priv->dev, "Failed to reset the dma, device will work with reduced throughput\n");
+		ret = 0;
 	}
 
 	/* DMA Configuration */
-- 
Armbian

